library/cipher.c

```diff
@@ -401,6 +413,12 @@ int mbedtls_cipher_setkey(mbedtls_cipher_context_t *ctx,
         return mbedtls_cipher_get_base(ctx->cipher_info)->setkey_dec_func(ctx->cipher_ctx, key,
                                                                           ctx->key_bitlen);
     }
+#else
+    if (operation == MBEDTLS_ENCRYPT || operation == MBEDTLS_DECRYPT) {
```

Comment:
It seems to me we should return `UNSUPPORTED` when `DECRYPT` is requested with ECB, though.

Reply:
Yeah, in ECB mode we want encrypt to work and decrypt to return `NOT_SUPPORTED`.